### PR TITLE
Created onboarding screens that only show up when app is opened for the very first time

### PR DESCRIPTION
**Pull request by D-Ajay-Kumar**
…he first time

## Related Issue 

- The task was to create onboarding screens to give instructions to the user as to how to use the app's features.

Closes: #47 

### Describe the changes you've made

1. Created a set of onboarding screens with not much emphasis on the UI currently.
2. This set of onboarding screens appear only when the app is opened for the first time.
3. To check whether the app is opened the first time or not HiveDB is used to store the check.

## Type of change

What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behaviour of your code (Write NA if not)
NA

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: lib/pages/onboarding_screen.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+import 'package:introduction_screen/introduction_screen.dart';
+import 'package:safety/pages/login_page.dart';
+
+class OnboardingScreen extends StatelessWidget {
+  final List<PageViewModel> listPageViewModels = [
+    PageViewModel(
```
Comment (hiashutoshsingh) [inline lib/pages/onboarding_screen.dart:8]: Let me provide you with screenshots that will go here.

File: lib/pages/onboarding_screen.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return Scaffold(
+      body: IntroductionScreen(
+        pages: listPageViewModels,
+        done: const Text(
+          "Continue",
+          style: TextStyle(fontWeight: FontWeight.w600),
+        ),
+        onDone: () {
+          Navigator.pushReplacement(
+            context,
```
Comment (hiashutoshsingh) [inline lib/pages/onboarding_screen.dart:40]: Navigation should be done using our routes mechanism. Please look at another place how we have done the project.

File: lib/ui/splash.dart [hunk 3]
```diff
@@ -14,12 +16,37 @@ class Splash extends StatefulWidget {
 }
 
 class _SplashState extends State<Splash> {
+  Box<bool> firstTimeBox;
+  bool firstTime;
+
+  Future<void> getHiveDate() async {
```
Comment (hiashutoshsingh) [inline lib/ui/splash.dart:22]: Please add comments here on what exactly this piece of code is doing.

File: lib/ui/splash.dart [hunk 4]
```diff
@@ -14,12 +16,37 @@ class Splash extends StatefulWidget {
...
-    });
+
+    getHiveDate();
+
+    Timer(
+      Duration(seconds: 3),
+      () {
+        Navigator.pushReplacement(
+          context,
+          MaterialPageRoute(
```
Comment (hiashutoshsingh) [inline lib/ui/splash.dart:38]: Use Routes Navigation mechanism

**Review by hiashutoshsingh — CHANGES_REQUESTED** [review]
I have added some comments which need a little change. Not some major changes though.

Let me know if something is not explained properly.

File: lib/pages/onboarding_screen.dart (hunk 2 above)
Comment (D-Ajay-Kumar) [reply lib/pages/onboarding_screen.dart:40]: Yeah I saw you guys are using GetIt for navigation

File: lib/pages/onboarding_screen.dart (hunk 1 above)
Comment (D-Ajay-Kumar) [reply lib/pages/onboarding_screen.dart:8]: sure

**Comment by hiashutoshsingh** [comment]
@D-Ajay-Kumar Please also resolve this conflict.

**Comment by D-Ajay-Kumar** [comment]
> @D-Ajay-Kumar Please also resolve this conflict.

Resolved the conflict, also added comments and changed routing mechanism. Could you provide with the screenshots of the onboarding screens or maybe create another issue for redesigning the screens according to provided pictures for other beginners?

**Comment by hiashutoshsingh** [comment]
> > @D-Ajay-Kumar Please also resolve this conflict.
> 
> Resolved the conflict, also added comments and changed routing mechanism. Could you provide with the screenshots of the onboarding screens or maybe create another issue for redesigning the screens according to provided pictures for other beginners?

Sure.
@shagun25 Please provide him screenshots and he is blocked on this.

**Comment by hiashutoshsingh** [comment]
@D-Ajay-Kumar hey, please take screenshots of the app from your side and place them in assets and use that.

**Comment by D-Ajay-Kumar** [comment]
@hiashutoshsingh I am afraid I don't follow. I need to take screenshots of what screens of the app and use them as onboarding screens?

**Comment by hiashutoshsingh** [comment]
> @hiashutoshsingh I am afraid I don't follow. I need to take screenshots of what screens of the app and use them as onboarding screens?

Sorry I wasn't clear enough.
Take screenshots of homescreen, sos, videos section, camera section..etc (if more can be added) from our app only.
so that user can know what kind of features are there in app.
@D-Ajay-Kumar

**Comment by hiashutoshsingh** [comment]
@D-Ajay-Kumar Hey, any update on the above?

**Comment by D-Ajay-Kumar** [comment]
Sorry about the delay in work. My college has announced surprise tests all of a sudden so I'm having direct most of my time there. Would be busy with that for 2 more days.

**Comment by hiashutoshsingh** [comment]
> Sorry about the delay in work. My college has announced surprise tests all of a sudden so I'm having direct most of my time there. Would be busy with that for 2 more days.

ohhh OK. not an issue.

**Comment by hiashutoshsingh** [comment]
@D-Ajay-Kumar any updates on the above, this is pending for a long.

Can you please resolve conflicts and attach screenshots.

Thanks

**Pull request closed without merging**